File: packages/core/rules/test/excessive-agency/positive.py
# Test cases that SHOULD be detected by excessive-agency rules

import os
import subprocess
import shutil

# mcp-python-automatic-deletion - should detect
os.remove(file_path)
os.unlink(file_path)
shutil.rmtree(dir_path)

# mcp-python-automatic-execution - should detect
os.system(user_command)
subprocess.call(user_command)

# mcp-python-unrestricted-write - should detect
with open(user_path, "w") as f:
    f.write(user_data)

File: packages/core/rules/test/excessive-agency/positive.js
// Test cases that SHOULD be detected by excessive-agency rules

const fs = require('fs');
const { exec, spawn } = require('child_process');

// mcp-automatic-file-deletion - should detect
fs.unlinkSync(filePath);
fs.rm(dirPath, { recursive: true });

// mcp-automatic-code-execution - should detect
exec(userCommand);
spawn(userCommand);

// mcp-unrestricted-network-access - should detect
async function myTool(url) {
  const data = await fetch(url);
  return data;
}

// mcp-unrestricted-file-write - should detect
fs.writeFileSync(userPath, userData);
